packages/sit-onyx/src/composables/useMoreList.ts

```diff
@@ -56,98 +44,113 @@ export type UseMoreListOptions = {
+   */
+  const memorizedComponentWidthMap = ref(new Map<string, number>());
+
+  // sync widthMap to components but keep component width if it changes to zero (e.g. because the element is hidden)
+  // so we can continue calculating if hidden elements would fit into the available width if resized
+  watchEffect(() => {
+    componentMap.value.forEach((value, key) => {
+      const newWidth = unref(value.width);
+      if (memorizedComponentWidthMap.value.has(key) && newWidth === 0) return;
+      memorizedComponentWidthMap.value.set(key, newWidth);
+    });
+  });
 
-          changeEntries.forEach((entry) => {
-            const elementId = Array.from(options.componentRefs).find(([_, element]) => {
-              return getTemplateRefElement(unref(element)) === entry.target;
-            })?.[0];
-            if (!elementId) return;
+  onMounted(() => {
+    watchEffect(() => {
+      let availableWidth = parentWidth.value;
+      if (availableWidth <= 0) return; // parent width is not initialized yet
 
-            const isFullyVisible = entry.intersectionRatio === 1;
+      const parentGap = getColumnGap(options.parentRef.value);
+      const listGap = getColumnGap(options.listRef.value);
 
-            if (isFullyVisible) shownIds.push(elementId);
-            else hiddenIds.push(elementId);
-          });
+      if (moreIndicatorWidth.value > 0) {
+        availableWidth -= moreIndicatorWidth.value + parentGap;
+      }
 
-          if (visibleElements.value.length === 0) {
-            visibleElements.value = shownIds;
+      // calculate which components currently fully fit into the available parent width
+      const { visible, hidden } = Array.from(memorizedComponentWidthMap.value.entries()).reduce(
+        (acc, [id, componentWidth], index) => {
+          availableWidth -= componentWidth + (index > 0 ? listGap : 0);
+
+          if (
+            availableWidth >= 0 ||
+            // check if last element fits if more indicator would be hidden
+            (index === memorizedComponentWidthMap.value.size - 1 &&
+              availableWidth + moreIndicatorWidth.value >= 0)
+          ) {
+            acc.visible.push(id);
           } else {
-            visibleElements.value = visibleElements.value
-              // remove now hidden elements
-              .filter((id) => !hiddenIds.includes(id))
-              // add newly visible elements
-              .concat(shownIds);
+            acc.hidden.push(id);
           }
+
+          return acc;
         },
-        { root, threshold: 1 },
+        { visible: [] as string[], hidden: [] as string[] },
       );
 
-      options.componentRefs.forEach((ref) => {
-        const element = getTemplateRefElement(unref(ref));
-        if (!element) return;
-        observer.value?.observe(element);
-      });
-    },
-    { immediate: true },
-  );
+      visibleElements.value = visible;
+      hiddenElements.value = hidden;
+    });
```

Comment:
Without `reduce` its a bit simpler imho:
```suggestion
      visibleElements.value = [];
      hiddenElements.value = [];

      Array.from(memorizedComponentWidthMap.value.entries()).forEach(
        ([id, componentWidth], index) => {
          availableWidth -= componentWidth + (index > 0 ? listGap : 0);

          if (
            availableWidth >= 0 ||
            // check if last element fits if more indicator would be hidden
            (index === memorizedComponentWidthMap.value.size - 1 &&
              availableWidth + moreIndicatorWidth.value >= 0)
          ) {
            visibleElements.value.push(id);
          } else {
            hiddenElements.value.push(id);
          }
        },
      );
```

We don't need to worry here about changing accessing the ref multiple times, as Vue batches changes.